front/lib/upsert_document.ts

```diff
@@ -1,28 +1,44 @@
-import type { Result } from "@dust-tt/types";
-import { Err, FrontDataSourceDocumentSection } from "@dust-tt/types";
+import type {
+  CoreAPIDocument,
+  CoreAPILightDocument,
+  DataSourceType,
+  Result,
+  UpsertContext,
+} from "@dust-tt/types";
+import {
+  Err,
+  FrontDataSourceDocumentSection,
+  sectionFullText,
+  UpsertContextSchema,
+} from "@dust-tt/types";
 import { Storage } from "@google-cloud/storage";
 import * as t from "io-ts";
 import { v4 as uuidv4 } from "uuid";
 
+import { getDocumentsPostUpsertHooksToRun } from "@app/documents_post_process_hooks/hooks";
+import { launchRunPostUpsertHooksWorkflow } from "@app/documents_post_process_hooks/temporal/client";
+import { Authenticator } from "@app/lib/auth";
 import { launchUpsertDocumentWorkflow } from "@app/upsert_queue/temporal/client";
 
-const EnqueueUpsertDocument = t.type({
-  projectId: t.string,
+const { DUST_UPSERT_QUEUE_BUCKET, SERVICE_ACCOUNT } = process.env;
+
+export const EnqueueUpsertDocument = t.type({
   workspaceId: t.string,
   dataSourceName: t.string,
   documentId: t.string,
   tags: t.union([t.array(t.string), t.null]),
   parents: t.union([t.array(t.string), t.null]),
   sourceUrl: t.union([t.string, t.null]),
+  timestamp: t.union([t.number, t.null]),
```

Comment:
Can we put `timestampMs`?

Reply:
We use timestamp everywhere for Data Source documents so I'd be in favor of valuing consistency over precision here?